nagios_test.go

```diff
@@ -0,0 +1,29 @@
+package main
+import (
+	"testing"
+)
+
+
+func TestGetNagiosTasksErrorWithMissingNagiosPod(t *testing.T) {
+	tasks := make(chan Task, 1)
+	GetNagiosTasks(tasks, []string{"project1"}, "basepath", func(project, resource, substr string) ([]string, error) {
+		return []string{}, nil
+	})
+	close(tasks)
```

Comment:
This block is not properly synchronized. There's no guarantee here that `GetNagiosTasks` will send any task before the channel is closed, and sending on a closed channel panics.